Geth attach no longer works

```
jbrown@akalabeth:~/go-ethereum$ build/bin/geth attach
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0xb0 pc=0x60edf0]

goroutine 1 [running]:
github.com/ethereum/go-ethereum/rpc/api.ParseApiString(0xc820252180, 0x35, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/rpc/api/utils.go:168 +0xb60
main.(*jsre).apiBindings(0xc820236ee0, 0x7f95041db788, 0xc820236ee0, 0x0, 0x0)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/js.go:279 +0x28f
main.newLightweightJSRE(0xc820250930, 0x1, 0x7f95041da570, 0xc820252000, 0xc820197f00, 0x16, 0x16f3901, 0x0)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/js.go:156 +0x16c
main.attach(0xc82019d560)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:440 +0x3fb
github.com/codegangsta/cli.Command.Run(0xe7abb0, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0xfb4e20, 0x42, 0x10b19a0, ...)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/codegangsta/cli/command.go:131 +0x1055
github.com/codegangsta/cli.(*App).Run(0xc82019d320, 0xc82000a440, 0x2, 0x2, 0x0, 0x0)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/codegangsta/cli/app.go:175 +0xfd6
main.main()
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/cmd/geth/main.go:372 +0x93

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/lib/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
	/usr/lib/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/lib/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 6 [chan receive]:
github.com/ethereum/go-ethereum/logger/glog.(*loggingT).flushDaemon(0x16f5340)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:897 +0x67
created by github.com/ethereum/go-ethereum/logger/glog.init.1
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/glog/glog.go:428 +0x6f

goroutine 18 [select]:
github.com/ethereum/go-ethereum/logger.dispatchLoop()
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/sys.go:81 +0x624
created by github.com/ethereum/go-ethereum/logger.init.1
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/logger/sys.go:60 +0x2b

goroutine 21 [select]:
github.com/ethereum/go-ethereum/jsre.(*JSRE).runEventLoop(0xc820246660)
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/jsre/jsre.go:139 +0xd18
created by github.com/ethereum/go-ethereum/jsre.New
	/home/jbrown/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/jsre/jsre.go:67 +0x139
```

--- a/rpc/api/admin.go
+++ b/rpc/api/admin.go
@@ -32,7 +32,6 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/eth"
 	"github.com/ethereum/go-ethereum/logger/glog"
-	"github.com/ethereum/go-ethereum/p2p"
 	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/ethereum/go-ethereum/rpc/codec"
 	"github.com/ethereum/go-ethereum/rpc/comms"
@@ -81,17 +80,15 @@ type adminhandler func(*adminApi, *shared.Request) (interface{}, error)
 // admin api provider
 type adminApi struct {
 	xeth     *xeth.XEth
-	network  *p2p.Server
 	ethereum *eth.Ethereum
 	codec    codec.Codec
 	coder    codec.ApiCoder
 }
 
 // create a new admin api instance
-func NewAdminApi(xeth *xeth.XEth, network *p2p.Server, ethereum *eth.Ethereum, codec codec.Codec) *adminApi {
+func NewAdminApi(xeth *xeth.XEth, ethereum *eth.Ethereum, codec codec.Codec) *adminApi {
 	return &adminApi{
 		xeth:     xeth,
-		network:  network,
 		ethereum: ethereum,
 		codec:    codec,
 		coder:    codec.New(nil),
@@ -140,11 +137,11 @@ func (self *adminApi) AddPeer(req *shared.Request) (interface{}, error) {
 }
 
 func (self *adminApi) Peers(req *shared.Request) (interface{}, error) {
-	return self.network.PeersInfo(), nil
+	return self.ethereum.Network().PeersInfo(), nil
 }
 
 func (self *adminApi) NodeInfo(req *shared.Request) (interface{}, error) {
-	return self.network.NodeInfo(), nil
+	return self.ethereum.Network().NodeInfo(), nil
 }
 
 func (self *adminApi) DataDir(req *shared.Request) (interface{}, error) {

--- a/rpc/api/utils.go
+++ b/rpc/api/utils.go
@@ -165,7 +165,7 @@ func ParseApiString(apistr string, codec codec.Codec, xeth *xeth.XEth, eth *eth.
 	for i, name := range names {
 		switch strings.ToLower(strings.TrimSpace(name)) {
 		case shared.AdminApiName:
-			apis[i] = NewAdminApi(xeth, eth.Network(), eth, codec)
+			apis[i] = NewAdminApi(xeth, eth, codec)
 		case shared.DebugApiName:
 			apis[i] = NewDebugApi(xeth, eth, codec)
 		case shared.DbApiName:

